src/main/java/cpw/mods/modlauncher/ClassTransformer.java

```diff
@@ -121,7 +121,8 @@
         finalFlags &= ~ILaunchPluginService.ComputeFlags.SIMPLE_REWRITE; //Strip any modlauncher-custom fields
 
         //Only use the TransformerClassWriter when needed as it's slower, and only COMPUTE_FRAMES calls getCommonSuperClass
-        ClassWriter cw = finalFlags == ILaunchPluginService.ComputeFlags.COMPUTE_FRAMES ? new TransformerClassWriter(this, clazz) : new ClassWriter(finalFlags | Opcodes.ASM7);
+        boolean requireFrames = (finalFlags & ILaunchPluginService.ComputeFlags.COMPUTE_FRAMES) == ILaunchPluginService.ComputeFlags.COMPUTE_FRAMES;
+        ClassWriter cw = requireFrames ? new TransformerClassWriter(this, clazz) : new ClassWriter(finalFlags | Opcodes.ASM7);
```

Comment:
As a style choice, why have this switch external in the first place? Would it not make more sense to just pass the flags as-computed to `TransformerClassWriter` and have that class then selectively enable the functionality required. This also means:

* The additional compute flag constants can be simply defined in `TransformerClassWriter` where they belong, much like the flags for `ClassWriter`
* The functionality for turning on/off classwriter features is naturally included in the classwriter used, instead of the ClassTransformer having to handle logic outside its remit. This means functionality added in the future (eg. `TransformerClassWriter.DUMP_FRAMES_TO_CONSOLE`) can then be logically added in the same place.

I get why you've done the "switch between the original classwriter and the enhanced classwriter" thing, but it seems like a contract which should be neatly encapsulated in the classwriter is leaking. If you move the flag handling into `TransformerClassWriter` then the domains of operations simply become:

* ClassTransformer collects flags
* TransformerClassWriter consumes flags